HOWTO-make-a-release.md

```diff
@@ -0,0 +1,400 @@
+For more information, please read Github's [creating a security advisory],
+including the "Next Steps" at the end of that page.
+
+[creating a security advisory]:
+<https://docs.github.com/en/free-pro-team@latest/github/managing-security-vulnerabilities/creating-a-security-advisory>
+
+The release person and the reviewer are allowed to use other means to
+share the commits to be reviewed if they desire.
+
+The release person and the reviewer must have a conversation to confirm
+or figure out how review shall be done.
+
+# Pre-publishing tasks
+
+Some of the actions in this section need to be repeated for each OpenSSL
+version released.
+
+## Freeze the source repository
+
+The day before the release, freeze the main repository.  This locks
+out everyone but the named user, who is doing the release, from doing
+any pushes.  Someone other than the person doing the release should
+run the command.  For example:
+
+    ssh openssl-git@git.openssl.org freeze openssl NAME
+
+## Prepare your repository checkouts
+
+You will need to checkout at least three working trees:
+
+-   one for the website
+
+        git clone openssl-git@git.openssl.org:openssl-web.git website
+
+-   one for extra tools
+
+        git clone openssl-git@git.openssl.org:tools.git tools
+
+    The resulting directory will be referred to as `$TOOLS`
+
+-   At least one for openssl source
+
+        git clone openssl-git@git.openssl.org:openssl.git
+
+    If you're doing multiple releases in one go, there are many ways to deal
+    with it.  One possibility, available since git 2.5, is to use `git
+    worktree`:
+
+        (cd openssl;
+         git worktree add ../openssl-1.1.1 OpenSSL_1_1_1-stable)
+
+## Make sure that the openssl source is up to date
+
+The person doing the release and the reviewer should both sanity-check
+the source to be released at this point.  Checks to consider include
+building and verify that make test passes on multiple plaforms - Linux,
+Windows, etc.
+
+For each source checkout, make sure that the CHANGES.md / CHANGES and
+NEWS.md / NEWS files have been updated and reviewed.
+
+NEWS.md / NEWS should contain a summary of any changes for the release,
+and for a security release is.  You should also update NEWS in the master
+branch to include details of all releases.  Just update the NEWS bullet
+points - do not change the release date, keep it as **under development**.
+
+Add any security fixes to the tree and commit them.
+
+Make sure that the copyrights are updated.  This script will update
+the copyright markers and commit the changes (where $TOOLS stands for
+the openssl-tools.git checkout directory):
+
+    $TOOLS/release-tools/do-copyright-year
+
+Obtain approval for these commits from the reviewer and add the
+reviewed-by headers as required.
+
+*Do* send the auto-generated commits to the reviewer and await their
+approval.
+
+*Do not push* changes to the main source repo at this stage, if this is a
+secturity release.
```

Comment:
I would drop "if this is a security release". There is no need to do things differently for a non-security release vs a security release.

Reply:
Is there any reason to hold back the push when *not* a security release?

Reply:
We have never done so previously, so that would be a change. I think it is simpler to keep the process between security vs normal releases as close as possible. I don't see a reason to have them different here.

Reply:
Besides, there is a typo 'secturity'

Reply:
[sigh] I give in, at least for expedience